concordium-consensus/src/Concordium/KonsensusV1/Types.hs

```diff
@@ -0,0 +1,804 @@
+{-# LANGUAGE BinaryLiterals #-}
+{-# LANGUAGE DeriveTraversable #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeFamilies #-}
+
+module Concordium.KonsensusV1.Types where
+
+import Control.Monad
+import Data.Bits
+import qualified Data.ByteString as BS
+import qualified Data.Map.Strict as Map
+import Data.Serialize
+import qualified Data.Vector as Vector
+import Data.Void
+import Data.Word
+import Numeric.Natural
+
+import qualified Concordium.Crypto.BlockSignature as BlockSig
+import qualified Concordium.Crypto.BlsSignature as Bls
+import qualified Concordium.Crypto.SHA256 as Hash
+import qualified Concordium.Crypto.VRF as VRF
+import Concordium.Types
+import Concordium.Types.HashableTo
+import Concordium.Types.Transactions
+import Concordium.Utils.Serialization
+
+import qualified Concordium.GlobalState.Basic.BlockState.LFMBTree as LFMBT
+
+-- |A round number for consensus.
+newtype Round = Round {theRound :: Word64}
+    deriving (Eq, Ord, Show, Serialize, Num, Integral, Real, Enum, Bounded)
+
+-- |A strict version of 'Maybe'. We deliberately avoid defining generalised serialization and
+-- hashing instances, so that specific instances can be given as appropriate.
+data Optionally a
```

Comment:
Is there a particular reason to not use the name that is standard in many other places, e.g., Ocaml, Rust, i.e., `Option`? It is the same as those.

Reply:
Renamed to Option.